### DISCUSSION TITLE
Vb.net Challenge Reading Private Tags

### DISCUSSION THREAD

**Discussion by bc2409** [fo-dicom/fo-dicom#1440] [Q&A] [answered]
Hello All,

Looking for some assistance in reading private tags.  Here is some relevant information:

VB.net on .Net Framework 4.7.2
fo-dicom version: 5.0.3.0

Here is a code snippet that I am having trouble with:
    
```
Try
            Dim openedDicom As DicomFile = DicomFile.Open("C:\temp\sr.dcm", FileReadOption.ReadAll)

            Dim nTag = New DicomTag(217, 1212, "nil.lexmark.com")
            Dim TString As String = openedDicom.Dataset.GetSingleValue(Of String)(nTag)
            Return TString
Catch ex As Exception
            Console.WriteLine(ex.Message)
End Try
```

Attached/below is a screenshot with the private tag I am trying to retrieve the value for (the last tag 0217,1212).

![Screen Shot 2022-09-06 at 5 15 56 PM](https://user-images.githubusercontent.com/84942488/188761677-f9bc3484-7a7f-4a9b-a213-bb77d1fb07a9.png)

The error I keep getting is: Exception thrown: 'FellowOakDicom.DicomDataException' in fo-dicom.core.dll
Tag: (00d9,04bc:nil.lexmark.com) not found in dataset

Thanks for any guidance.

**Comment by mrbean-bremen** [comment, marked as answer]
Looks like you are using decimal instead of hex values, you need something like:
```vb
Dim nTag = New DicomTag(&H0217, &H1212, "nil.lexmark.com")
```
DICOM tags are always represented as hexadecimal values.
